core/primitives/src/shard_layout.rs

```diff
@@ -343,6 +342,7 @@ impl std::error::Error for ShardLayoutError {}
 
 impl ShardLayout {
     /// Handy constructor for a single-shard layout, mostly for test purposes
+    #[cfg(feature = "sample_shard_layout")]
```

Comment:
Can you instead reuse one of the existing features? Either `test` or `test_utils` seem reasonable.